auth/middleware.go

```
@@ -55,5 +58,25 @@ func middleware(c *gin.Context, db *sql.DB) {
 		c.AbortWithStatusJSON(403, gin.H{"message": "authentication failed"})
 		return
 	}
+	c.Set(userContextKey, user)
+	c.Set(coverageContextKey, coverage)
 	c.Next()
 }
+
+//return the user associated to the context if no user has been set it returns (User{}, false)
+func GetUser(c *gin.Context) (user User, ok bool) {
+	tmp, ok := c.Get(userContextKey)
+	if !ok {
+		return User{}, ok
+	}
+	return tmp.(User), ok
```

can't this silently fail?

---

It will panic if it isn't the right type: https://tour.golang.org/methods/15